runtime/interpreter/decode.go

```diff
@@ -445,7 +431,7 @@ func (d *DecoderV4) decodeDictionary(path []string) (*DictionaryValue, error) {
 		deferred = orderedmap.NewStringStructOrderedMap()
 		deferredOwner = d.owner
 		deferredStorageKeyBase = joinPath(append(path, dictionaryValuePathPrefix))
-		for _, keyValue := range keys.Values {
+		for _, keyValue := range keys.Elements() {
```

Comment:
What do you think about not deferring decoding of keys?  Keys are accessed in `decodeDictionary`, so deferring decoding of keys can add overhead.

Reply:
Yes, we can definitely improve the dictionaries as well. I had added that to epic #844.
We could do that in a separate PR as the next imrpovement.

Reply:
I see. That's a good point. Yup, turning off the deferring, might be a good idea. Thanks for catching that!

Reply:
Added with 3f96789